### PR TITLE
feat(performance): Add queues module tables components

### PR DESCRIPTION
**Pull request by edwardgou-sentry**
Adds Queues module table components with tests

### CODE REVIEW

File: static/app/views/performance/queues/destinationSummary/transactionsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+import useOrganization from 'sentry/utils/useOrganization';
+import {normalizeUrl} from 'sentry/utils/withDomainRequired';
+import {renderHeadCell} from 'sentry/views/starfish/components/tableCells/renderHeadCell';
+import {useSpanMetrics} from 'sentry/views/starfish/queries/useSpanMetrics';
+import type {MetricsResponse} from 'sentry/views/starfish/types';
+import {QueryParameterNames} from 'sentry/views/starfish/views/queryParameters';
+
+type Row = Pick<
+  MetricsResponse,
+  | 'avg_if(span.self_time,span.op,queue.task.celery)'
```
Comment (gggritso) [inline static/app/views/performance/queues/destinationSummary/transactionsTable.tsx:29]: 👀 this is interesting, why? Do queue span ops not map cleanly to a module? Why is the average `avg_if`, but the `sum` isn't a `sum_if`?

File: static/app/views/performance/queues/destinationSummary/transactionsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+  const location = useLocation();
+  const destination = decodeScalar(location.query.destination);
+  const cursor = decodeScalar(location.query?.[QueryParameterNames.DOMAINS_CURSOR]);
+
+  const mutableSearch = new MutableSearch(
+    'span.op:[queue.task.celery,queue.submit.celery]'
+  );
+  if (destination) {
+    mutableSearch.addFilterValue('transaction', destination);
+  }
```
Comment (gggritso) [inline static/app/views/performance/queues/destinationSummary/transactionsTable.tsx:99]: IMO it would be a good idea to use the name "transaction" everywhere except the user-facing label. If the UI label for a transaction is "destination" that's  totally fine, but if under the hood destinations are just transactions, the code should talk about transactions! Otherwise it's confusing

File: static/app/views/performance/queues/messageSpanSamplesTable.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  jest.mocked(useOrganization).mockReturnValue(organization);
+
+  beforeEach(() => {});
+  it('renders', () => {
+    render(<MessageSpanSamplesTable data={[]} isLoading={false} />);
+    screen.getByText('Span ID');
+    screen.getByText('Message ID');
+    screen.getByText('Processing Latency');
+    screen.getByText('Message Size');
+    screen.getByText('Status');
```
Comment (gggritso) [inline static/app/views/performance/queues/messageSpanSamplesTable.spec.tsx:21]: If you check the HTTP module specs for inspo, there are some nice tests in there that check for a `table` role, and `cell` roles to make sure there's a table 👍🏻 Up to you, though

**Review by gggritso — APPROVED** [review]
Makes sense! Maybe worth renaming `destination` to `transaction` in the code, and adding more specific specs for tables and table cells. Also, check out the `http` module folder structure! Tables and charts have their own directories which helps if tables and charts are shared. Totally up to you, though

File: static/app/views/performance/queues/messageSpanSamplesTable.spec.tsx (hunk 3 above)
Comment (edwardgou-sentry) [reply static/app/views/performance/queues/messageSpanSamplesTable.spec.tsx]: Nice, I've updated the tests to check by role!

File: static/app/views/performance/queues/destinationSummary/transactionsTable.tsx (hunk 2 above)
Comment (edwardgou-sentry) [reply static/app/views/performance/queues/destinationSummary/transactionsTable.tsx:100]: So this filter should actually be filtering by a `destination` tag, not transaction. Since `destination` isn't yet tagged or implemented in the sdks, I'm using transaction temporarily to continue building out the product UI. Added a TODO comment around this to be more clear about this situation!

File: static/app/views/performance/queues/destinationSummary/transactionsTable.tsx (hunk 1 above)
Comment (edwardgou-sentry) [reply static/app/views/performance/queues/destinationSummary/transactionsTable.tsx:30]: So right now `queue.task.celery` ops refer to consumer spans and `queue.submit.celery` refers to producer spans (for celery).
 
One thing is, I haven't added `queue` spans as a known module yet in relay, which is why I'm not able to query by `span.module`. 

The other thing is that for this query, I'm filtering this query to include both `task` and `submit` ops, but i just want the average duration of the `task` ops as a column. The `sum` in this query is purposefully the total duration sum across both `task` and `submit` which is why I don't need any conditional `sum_if` here (although this sum will probably change because it's not really conveying what I'm looking for either. That's a different topic).

So basically this is because I'm still experimenting with queue spans in ui and relay at the moment 😄

Comment (gggritso) [reply static/app/views/performance/queues/destinationSummary/transactionsTable.tsx:30]: Roger

File: static/app/views/performance/queues/destinationSummary/transactionsTable.tsx (hunk 2 above)
Comment (gggritso) [reply static/app/views/performance/queues/destinationSummary/transactionsTable.tsx:100]: Ah, okay!
